agent-backend/src/build/build_crew.py

```diff
@@ -191,8 +108,12 @@ def build_tools_and_their_datasources(self):
             for model_key, model_object in tool_models_objects.items():
                 model_data = match_key(self.models_models, model_key)
                 tool_models_models.append((model_object, model_data))
+
+            # Use same llm as Crew's for retriever
+            retriever_llm = match_key(self.crew_models, keyset(self.crew_model.id))
```

Comment:
The self-query retriever needs an LLM... for now I decided to pass the same LLM linked to the Crew. But we have to either take it as a user input or use one of the already configured models linked to Agent or Crew (as it is now). Let me know your thoughts!

Reply:
For now yes, the crew or "manager" llm is fine :+1: